packages/core/src/plugins.ts

```diff
@@ -0,0 +1,50 @@
+import { Character, DynamicPlugin, Plugin } from "./types";
+
+/**
+ * Gets a secret from the character settings or environment variables
+ * @param {Character} character - The character object
+ * @param {string} secret - The secret key
+ * @returns {string|undefined} The secret value or undefined if not found
+ */
```

Comment:
nit: personally would remove type information in the JSDocs since its completely redudant when we are using typescript